test/es-module/test-typescript-commonjs.mjs

```diff
@@ -59,13 +59,43 @@ test('require a .ts file with implicit extension fails', async () => {
 });
 
 test('expect failure of an .mts file with CommonJS syntax', async () => {
-  const result = await spawnPromisified(process.execPath, [
-    fixtures.path('typescript/cts/test-cts-but-module-syntax.cts'),
-  ]);
-
-  strictEqual(result.stdout, '');
-  match(result.stderr, /To load an ES module, set "type": "module" in the package\.json or use the \.mjs extension\./);
-  strictEqual(result.code, 1);
+  const testFilePath = fixtures.path(
+    'typescript/cts/test-cts-but-module-syntax.cts'
+  );
+  const result = await spawnPromisified(process.execPath, [testFilePath]);
+
+  assert.strictEqual(result.stdout, '');
+
+  const expectedWarning = `Failed to load the ES module: ${testFilePath}. Make sure to set "type": "module" in the nearest package.json file or use the .mjs extension.`;
+
+  // First assertion: check that stderr contains a generic pattern for the ES module load failure.
+  // This ensures that at a minimum the error message structure is as expected.
+  assert.match(
+    result.stderr,
+    /Failed to load the ES module:.*test-cts-but-module-syntax\.cts/
+  );
+
+  // Second assertion: verify that stderr includes the complete expected warning message.
+  // We use a try/catch here to improve the error output, providing more context if the assertion fails.
+  // Having two assertions ensures that:
+  //   1. We first confirm the error message follows the expected general format.
+  //   2. We then check that the specific warning (with file path and message details) is present.
+  try {
+    assert.ok(
+      result.stderr.includes(expectedWarning),
+      `Expected stderr to include: ${expectedWarning}`
```

Comment:
It's still unclear why we need two assertions, it seems to me the second one can only pass if the first one also passes (i.e. the first one doesn't add any coverage). Am I missing something?

Reply:
purpose of using two assertions is to check the general format and the details of the error message separately. The first assertion verifies that the basic structure of the message is correct, while the second assertion checks that the expected warning is included in the output.

Reply:
OK, but how is it useful to know that the structure is correct if the warning is not what we expected anyway?

Reply:
It shows that the error mechanism is working, so only the warning text is off

Reply:
Can't you figure it out just by reading the error output of the second assertion though?

Reply:
I think No, the second assertion shows the text mismatch, but the first confirms the error is generated correctly overall.

Reply:
I don't get it, but ok, whatever. Anyway, shouldn't we put the "first assertion" inside the `catch` block, it seems to me there's no point to check it if the full warning is correctly generated

Reply:
thanks a lot for your support, now I moved it to in try-catch line